STEP 1 completion (#19)
meilisearch/strapi-plugin-meilisearch: README.md

> # MeiliSearch in Strapi Plugin 🔎

Reviewer: This is ok for the moment, but keep in mind it should have the same header as the other at the end :) I'm going to ask for a MS logo with Strapi :D

Author: https://github.com/meilisearch/strapi-plugin-meilisearch/issues/23

> Index your strapi Collections into a MeiliSearch instance. The plugin listens to modifications made on your collections and update MeiliSearch accordingly.

Reviewer: ```suggestion
Index your Strapi collections into a MeiliSearch instance. The plugin listens to modifications made on your collections and update MeiliSearch accordingly.
```

<p align="center">

  <a href="https://github.com/prettier/prettier"><img src="https://img.shields.io/badge/styled_with-prettier-ff69b4.svg" alt="Prettier"></a>
  <a href="https://github.com/meilisearch/meilisearch-js/blob/main/LICENSE"><img src="https://img.shields.io/badge/license-MIT-informational" alt="License"></a>
  <a href="https://app.bors.tech/repositories/28762"><img src="https://bors.tech/images/badge_small.svg" alt="Bors enabled"></a>
</p>

> [MeiliSearch](https://github.com/meilisearch/meilisearch) is a open-source and easy to use search engine.

Reviewer: Can you use the sentence we choose in all our integration READMEs :)
```txt
MeiliSearch is an open-source search engine. Discover what MeiliSearch is!
```

[Strapi](https://strapi.io/) is a backend CMS that makes creating and managing content easy.

## Usage during WIP

> Until this package is released on `NPM`, you can use it the following way"

Reviewer: ```suggestion
Until this package is released on `npm`, you can use it the following way:
```

### ⏳ Installation

To use this project you will need to clone it:

```
$ git clone git@github.com:meilisearch/strapi-plugin-meilisearch.git
$ cd strapi-plugin-meilisearch
```

Install all required dependencies:

> ```

Reviewer: ```suggestion
```bash
```

# with yarn

> yarn install

Reviewer: ```suggestion
$ yarn install
```

# with yarn

> npm install

Reviewer: ```suggestion
$ npm install
```

```

### Using playground

Instead of adding the plugin to an existing project, you can try it out using the playground.

> ```

Reviewer: ```suggestion
```bash
```

# with yarn

> yarn develop

Reviewer: ```suggestion
$ yarn develop
```

```

Or if you want to use `npm`, by going inside the directory:

> ```

Reviewer: ```suggestion
```bash
```

# with npm

> cd playground
> npm install
> npm run develop

Reviewer: ```suggestion
$ cd playground
$ npm install
$ npm run develop
```

```

Install Strapi with this **Quickstart** command to create a Strapi project instantly:

#### Create strapi project

> - (Use **yarn** to install the Strapi project (recommended). [Install yarn with these docs](https://yarnpkg.com/lang/en/docs/install/).)

Reviewer: ```suggestion
- Use **yarn** to install the Strapi project (recommended). [Install yarn with these docs](https://yarnpkg.com/lang/en/docs/install/).)
```

```bash
# with yarn

> yarn create strapi-app my-project --quickstart

Reviewer: ```suggestion
$ yarn create strapi-app my-project --quickstart
```

# with npm/npx

> npx create-strapi-app my-project --quickstart

Reviewer: ```suggestion
$ npx create-strapi-app my-project --quickstart
```

```

_This command generates a brand new project with the default features (authentication, permissions, content management, content type builder & file upload). The **Quickstart** command installs Strapi using a **SQLite** database which is used for prototyping in development._

> Once your strapi project has been created, to link the plugin to this project you have to create a symbolic link inside a plugin folder at the root of the strapi project.

Reviewer: ```suggestion
Once your Strapi project has been created, to link the plugin to this project you have to create a symbolic link inside a plugin folder at the root of the Strapi project.
```

1. Create plugin folder

> ```
> mkdir plugins
> ```
> 2. Create symbolic link
> ```
> cd plugins
> ln -s [PATH_TO_PLUGIN] meilisearch
> ```
> 3. Develop
> ```
> yarn develop
> ```

Reviewer: ```suggestion
```bash
$ mkdir plugins
```
2. Create symbolic link
```bash
$ cd plugins
$ln -s [PATH_TO_PLUGIN] meilisearch
```
3. Develop
```bash
$ yarn develop
```
```

> You can now use the plugin on your strapi project.

Reviewer: ```suggestion
You can now use the plugin on your Strapi project.
```

## 🖐 Requirements

> Complete installation requirements are exact same as for Strapi itself and can be found in the documentation under <a href="https://strapi.io/documentation/v3.x/installation/cli.html#step-1-make-sure-requirements-are-met">Installation Requirements</a>.

Reviewer: Why don't you use the markdown link? This is easier to read (for maintainer)

Author: It was a copy paste from another readme, changing that

Author: It was a copy paste from another strapi readme ! Changed it

## 🖐 Requirements

Complete installation requirements are exact same as for Strapi itself and can be found in the documentation under [Installation Requirements](https://strapi.io/documentation/v3.x/installation/cli.html#step-1-make-sure-requirements-are-met).

**Supported Strapi versions**:

- Strapi v3.4.x

(This plugin may work with the older Strapi versions, but these are not tested nor officially supported at this time.)

**Node / NPM versions**:

- NodeJS >= 12.10 <= 14
- NPM >= 6.x

**We recommend always using the latest version of Strapi to start your new projects**.

## 🌎 Community support

- For general help using Strapi, please refer to [the official Strapi documentation](https://strapi.io/documentation/).
- Strapi Slack [channel](https://slack.strapi.io/)

## Plugin's goal

> We want to create a Plugin that provides automatic actions and an interface on the strapi dashboard that helps the user start with MeiliSearch. The User should be able to create or remove new collections from MeiliSearch.

Reviewer: ```suggestion
We want to create a Pplugin that provides automatic actions and an interface on the Strapi dashboard that helps the user start with MeiliSearch. The User should be able to create or remove new collections from MeiliSearch.
```

> In the background, the plugin should re-index the documents that have been changed in the collections.
> For exemple, If I change the name of the restaurant `Tonio` with `Tony` in a strapi collection, an update should automaticly be done to update it aswell on MeiliSearch.

Reviewer: ```suggestion
In the background, the plugin should re-index the documents that have been changed in the collections.<br>
For example, If I change the name of the restaurant `Tonio` with `Tony` in a Strapi collection, an update should automatically be done to update it as well on MeiliSearch.
```

In the background, the plugin should re-index the documents that have been changed in the collections.<br>
For example, If I change the name of the restaurant `Tonio` with `Tony` in a Strapi collection, an update should automatically be done to update it as well on MeiliSearch.

No additional routes should be created on strapi for the front end users as they will be using the MeiliSearch API to search and not Strapi (unless we want to make this possible?).

## MVP

- Downloadable plugin
- Communication with a running MeiliSearch
- Interface with the following:
    - Add host and keys
    - Ability to chose which collections should be indexed in MeiliSearch
    - Every time a collection is updated, the updated element should be updated in MeiliSearch as well (listener to add, edit and delete)

>     - Possibility to remove a collection from MeiliSearch

Reviewer: I selected all the MVP part, not only the bonus one. I meant removing all the list and add a link to your issues that are certainly up to date :)

Author: Oh my bad

> - Possibility for the user to click `add to meilisearch` on Collection creation

Reviewer: Don't detail it here, it means maintaining it if we change our minds, redirect to the issues :) Issues will be up to date.
